monitoring/src/main/scala/akkaviz/server/WebSocketRepl.scala

```diff
@@ -0,0 +1,178 @@
+import akka.util.ByteString
+import ammonite.ops.Path
+import ammonite.repl._
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.concurrent.duration._
+import scala.concurrent.blocking
+
+trait WebSocketRepl {
+
+  def replPredef: String
+
+  def replArgs: Seq[Bind[_]]
+
+  def defaultReplPredef =
+    """
+      |import Predef.{println => _}
+      |import pprint.{pprintln => println}
+    """.stripMargin
+
+  implicit val ec = ExecutionContext.fromExecutor(Executors.newFixedThreadPool(4))
+
+  def replWebSocket: Route = {
+
+    def runRepl(replServerClassLoader: ClassLoader, in: InputStream, out: OutputStream) = {
+      val homePath = Path(Path.makeTmp)
+      val sshOut = new SshOutputStream(out)
+
+      Future {
+        val replSessionEnv = Environment(replServerClassLoader, in, sshOut)
+        Environment.withEnvironment(replSessionEnv) {
+          try {
+            blocking {
+              withConsoleRedirection {
+                val predef = defaultReplPredef + "\n" + replPredef
+                val repl = new Repl(in, sshOut, sshOut, Ref(Storage(homePath, None)), predef, replArgs)
+                repl.run()
+                repl
+              }
+            }
+          } catch {
+            case any: Throwable =>
+              val sshClientOutput = new PrintStream(sshOut)
+              sshClientOutput.println("What a terrible failure, the REPL just blow up!")
+              any.printStackTrace(sshClientOutput)
+              sshOut.flush()
+              sshOut.close()
+          }
+        }
+      }
+
+      def withConsoleRedirection(fun: => Any): Unit = {
+        Console.withIn(in) {
+          Console.withOut(sshOut) {
+            Console.withErr(sshOut) {
+              fun
+            }
+          }
+        }
+      }
+    }
+
+    val wsFlow: Flow[Message, Message, _] = {
+
+      val in = Flow[Message].flatMapConcat {
+        case b: BinaryMessage => b.dataStream
+        case t: TextMessage   => t.textStream.map(str => ByteString(str.getBytes))
+      }.toMat(StreamConverters.asInputStream(1.hour))(Keep.right)
+
+      val out = StreamConverters.asOutputStream(1.hour)
+        .groupedWithin(1000, 100.millis)
+        .map(_.reduce(_ ++ _))
+        .map[Message](bs => BinaryMessage.Strict(bs))
+
+      Flow.fromSinkAndSourceMat(in, out)(Keep.both).mapMaterializedValue {
+        case (in, out) => runRepl(this.getClass.getClassLoader, in, out)
+      }.keepAlive(30.seconds, () => BinaryMessage.Strict(ByteString()))
+
+    }
+
+    Directives.handleWebSocketMessages(wsFlow)
+
+  }
+
+  private class SshOutputStream(out: OutputStream) extends OutputStream {
+    override def close() = {
+      out.close()
+    }
+
+    override def flush() = {
+      out.flush()
+    }
+
+    override def write(byte: Int): Unit = {
+      if (byte.toChar == '\n') out.write('\r')
+      out.write(byte)
+    }
+
+    override def write(bytes: Array[Byte]): Unit = for {
+      i ← bytes.indices
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Non ascii characters are not allowed](https://www.codacy.com/app/lustefaniak/akka-viz/file/2101331339/issues/source?bid=3121282&fileBranchId=3142893#l108)